client/my-sites/email/email-management/home/email-mailbox-action-menu.jsx

```diff
@@ -57,15 +61,55 @@ const removeEmailForwardMailbox = ( { dispatch, mailbox } ) => {
 	dispatch( removeEmailForward( mailbox.domain, mailbox.mailbox ) );
 };
 
+const removeTitanMailbox = ( dispatch, mailbox, onAccountChanged, translate ) => {
+	recordTracksEvent( 'calypso_email_management_titan_delete_mailbox_dispatch', {
+		domain_name: mailbox.domain,
+		mailbox: mailbox.mailbox,
+	} );
+
+	const errorMessage = errorNotice(
+		translate( 'There was an error removing %(emailAddress)s from your account', {
+			comment: '%(emailAddress)s is the email address associated to the mailbox being deleted',
+			args: { emailAddress: getMailboxAddress( mailbox ) },
+		} )
+	);
+
+	const successMessage = successNotice(
+		translate( '%(emailAddress)s has been removed from your account', {
+			comment: '%(emailAddress)s is the email address associated to the mailbox being deleted',
+			args: { emailAddress: getMailboxAddress( mailbox ) },
+		} )
+	);
+
+	wp.undocumented()
+		.deleteTitanMailbox( mailbox.domain, mailbox.mailbox )
+		.then(
+			( response ) => {
+				if ( response.status === 202 ) {
+					onAccountChanged?.();
```

Comment:
I've never realized that we could use this new syntax for optional callbacks, nice.